packages/compiler/core/program.ts

```diff
@@ -620,7 +619,7 @@ export async function createProgram(
     }
 
     const expected = await host.realpath(
-      resolvePath(fileURLToPath(import.meta.url), "../index.js")
+      resolvePath(host.fileURLToPath(import.meta.url), "../index.js")
```

Comment:
is the adding `fileURLToPath` to the host needed? Wouldn't having the `realpath` handle the browser path deal with this?

Reply:
The root issue here is that vite doesn't have a browser shim for fileURLToPath, so it cannot be called at all for the browser. Adding it to host seemed like the easiest fix - the browser host has its own dumb implementation of this that works.

Reply:
Right this is not a a standard function, missed that.